include/revng/ADT/StringTemplateParameter.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <array>
+#include <string_view>
+
+template<size_t N>
+struct StringTemplateParameter {
```

Comment:
Can you put a couple of `static_assert`s to illustrate goals and usage?

Reply:
What do you have in mind?

As far as I see it, the only (non copy/move) construction accepting an array is a good enough guarantee that the object makes sense if it is constructed. Sure, it's possible to make runtime copies (for some reason), but then it's just a boring wrapper over the array with limited interface.

Reply:
And if you meant like example for other users to see, then it's probably better for them to just take a look where this thing is used, like at those scary string-accepting `std::get`s - definitely better than flat artificial examples.

Reply:
Yeah, but with no doc and no examples looking at it, it's not obvious how to use it.
An artificial example speaks more than a thousand words and saves you from `git grep`.

Reply:
Is this about what you had in mind: https://github.com/revng/revng/pull/197/commits/38409697f5862d86f20f2625aac04158092db5a8?